addons/gunbag/XEH_preInit.sqf

```diff
@@ -49,6 +49,9 @@ PREP_RECOMPILE_END;
     private _gunbagWeapon = _extendedInfo getOrDefault [QGVAR(gunbagWeapon), []];
     if (_gunbagWeapon isNotEqualTo []) then {
         (backpackContainer _unit) setVariable [QGVAR(gunbagWeapon), _gunbagWeapon, true];
+
+        // Prevent the arsenal closed event from overwriting new info
+        GVAR(arsenalCache) = nil;
```

Comment:
```suggestion
        // Prevent the arsenal closed event from overwriting new info
        if (!isNil QGVAR(arsenalCache)) then {
            GVAR(arsenalCache) = _gunbagWeapon;
        };
```

Reply:
Imo your suggestion is safer, committing.